### PR TITLE
Fix :  System Message: ERROR/3 (<string>, line 18) Unknown directive type "toctree". #537 

### PR DESCRIPTION
**Pull request by novicejava1**
This pull request is a fix to update the code to utilize sphinx for building the html documents instead of docutils which currently does not support the directive type "toctree" due to which there is a system error below the "Table of Contents" section.

### CODE REVIEW

File: datagrepper/app.py [hunk 1]
```diff
@@ -228,7 +192,7 @@ def index():
     total = count_all_messages()
     docs = load_docs(flask.request)
     return flask.render_template("index.html", docs=docs, total=total)
-
+
```
Comment (abompard) [inline datagrepper/app.py]: This looks like an oversight

File: datagrepper/app.py [hunk 2]
```diff
@@ -248,6 +212,12 @@ def widget():
     return flask.render_template("index.html", docs=load_docs(flask.request))
 
 
+@app.route("/contributing.html")
+def contributing():
+    print("In contributing call")
```
Comment (abompard) [inline datagrepper/app.py]: Looks like a debug leftover

File: devel/ansible/roles/datagrepper/tasks/main.yml [hunk 3]
```diff
@@ -13,6 +13,8 @@
       - libpq-devel
       - tox
       - python-psycopg2
+#      - libxml2-devel
+#      - libxslt-devel
```
Comment (abompard) [inline devel/ansible/roles/datagrepper/tasks/main.yml]: No need to add it if we don't need it.

**Review by abompard — CHANGES_REQUESTED** [review]
Thanks for working on this! Just a few leftovers I think.

File: datagrepper/app.py (hunk 1 above)
Comment (novicejava1) [reply datagrepper/app.py]: Fixed

File: datagrepper/app.py (hunk 2 above)
Comment (novicejava1) [reply datagrepper/app.py]: Fixed

File: devel/ansible/roles/datagrepper/tasks/main.yml (hunk 3 above)
Comment (novicejava1) [reply devel/ansible/roles/datagrepper/tasks/main.yml]: Fixed

**Comment by novicejava1** [comment]
HI @abompard , have fixed with the changes as per your review. Let me know if anything else..

**Comment by abompard** [comment]
Thanks! Please rebase and squash the commits. You can ignore the RPM build failure on Rawhide, that's expected.

**Comment by novicejava1** [comment]
Hi @abompard ,

This is how my toctreefix git log is looking right now. Could you help me understand on rebasing and squashing commits that i need to do.? 

* 31783fe (HEAD -> toctreefix, origin/toctreefix) Fixed app.py and ansible main.yml file as per review
* 7bda58c removing commented out line
* d9fa6b2 resolving merge conflict for poetry.lock
*   dd03ff8 resolving merge conflict for poetry.lock
|\  
| * 1bde52a (origin/develop, origin/HEAD) Bump incremental from 21.3.0 to 22.10.0
| * bcda86f Bump greenlet from 1.1.2 to 2.0.2
| * a701870 Bump blinker from 1.4 to 1.6.2
| * 6ec9ac2 Bump coverage from 6.5.0 to 7.2.7
| * a30f9d2 Bump automat from 20.2.0 to 22.10.0
| * ff7ce00 Fix packit config again
| * 70446b9 Bump mypy-extensions from 0.4.3 to 1.0.0
| * 0b8f3d7 Bump charset-normalizer from 2.1.1 to 3.1.0
| * d05d8be Fix the packit config file
| * 66deb01 Bump cryptography from 39.0.1 to 41.0.1
| * 96b0d07 Bump pyopenssl from 22.0.0 to 23.2.0
| * 74d40ba Bump arrow from 1.2.2 to 1.2.3
| * 5e719ea Switch from dependabot to renovate
| * 9e79e7e Auto-update pre-commit hooks
* | 4ad5aae Fix for  System Message: ERROR/3 (<string>, line 18) Unknown directive type "toctree". #537
|/  
* 7a1047b Bump requests from 2.28.2 to 2.31.0
* 9f258c2 Auto-update pre-commit hooks

**Comment by abompard** [comment]
Sure! What you could do is something like this, once you're on your `toctreefix` branch:
```
$ git fetch origin
$ git rebase -i origin/develop
```

This will show you all the changes you made since the current develop branch, prefixed by the letter `p`. Leave the `p` on the first one and change all the others to `f`. This will merge all your changes into the first commit, so you'll only have one left.
Then you can do `git push --force` and it will update your pull request.

**Comment by novicejava1** [comment]
Hi @abompard , Tried rebasing as per your suggesting. Was having issue with poetry.lock file while rebasing which i had to resolve the conflict. Please have a look.

**Comment by abompard** [comment]
Yeah, conflicts with `poetry.lock` are likely. Here's how I would resolve them:
- fix the conflicts in `pyproject.toml` (this one should be simple)
- remove poetry.lock
- run `poetry lock --no-update`
- run `git add poetry.lock && git rebase --continue` to finish the rebase process
- run `git push --force` to update the PR

**Pull request closed without merging**